doc/create-a-custom-rule.md

```diff
@@ -0,0 +1,242 @@
+function noGifAllowed(tree, file, options) {}
+```
+
+*   `tree` (*required*): [mdast](https://github.com/syntax-tree/mdast)
+*   `file` (*required*): [virtual file](https://github.com/vfile/vfile)
+*   `options` (*optional*): additional information passed to the rule by users
+
+[Back to Top](#contents)
+
+## Rule implementation
+
+Because we will be inspecting [mdast](https://github.com/syntax-tree/mdast), which is a markdown abstract syntax tree built upon [unist](https://github.com/syntax-tree/unist), we can take advantage of the many existing [unist utilities](https://github.com/syntax-tree/unist#utilities) to inspect our tree’s nodes.
+
+For this example, we will use [`unist-util-visit`](https://github.com/syntax-tree/unist-util-visit) to recursively inspect all the image nodes, and [`unist-util-generated`](https://github.com/syntax-tree/unist-util-generated) to ensure we are not inspecting nodes that we have generated ourselves and do not belong to the `doc.md`.
+
+```js
+const rule = require("unified-lint-rule");
+const visit = require("unist-visit-util");
+const generated = require("unist-util-generated");
+
+function isValidNode(node) {
+  // Here we check whether the given node violates our rule.
+  // Implementation details are not relevant to the scope of this example.
+  // This is an overly simplified solution for demonstration purposes
+  if (node.url && typeof node.url === "string") {
+    return !node.url.endsWith(".gif");
+  }
+}
+function noGifAllowed(tree, file, options) {
+  visit(tree, "image", visitor);
+  function visitor(node) {
+    if (!generated(node)) {
+       // This is an extremely simplified example of how to structure
+       // the logic to check whether a node violates your rule.
+       // You have complete freedom over how to visit/inspect the tree,
+       //and on how to implement the validation logic for your node.
+      const isValid = isValidNode(node);
+      if (!isValid) {
+        // Remember to provide the node as second argument to the message,
+        // in order to obtain the position and column where the violation occurred.
+        file.message(
+          `Invalid image file extentions. Please do not use gifs`,
+          node
+        );
+      }
+    }
+  }
+}
+module.exports = rule("remark-lint:no-gif-allowed", noGifAllowed);
+```
+
+[Back to Top](#contents)
+
+## Import the rule in your remark config
+
+Now that our custom rule is defined, and ready to be used, we need to add it to our `remark` configuration.
+
+You can do that by importing your rule and adding it in `plugins` array:
+
+```js
+// .remarkrc.js
+const noGifAllowed = require("./rules/no-gif-allowed.js");
+
+module.exports = {
+  plugins: [noGifAllowed],
+};
+```
+
+[Back to Top](#contents)
+
+## Apply the rule on the Markdown file
+
+If you run `npm lint`, you should see the following message in the terminal:
+
+```sh
+ 5:1-5:30  warning  Invalid image file extentions. Please do not use gifs  no-gif-allowed  remark-lint
```

Comment:
````suggestion
```text
5:1-5:30  warning  Invalid image file extentions. Please do not use gifs  no-gif-allowed  remark-lint
````